src/meta/src/barrier/checkpoint/creating_job/mod.rs

```diff
@@ -84,27 +88,24 @@ impl CreatingStreamingJobControl {
         );
         let fragment_infos: HashMap<_, _> = info.stream_job_fragments.new_fragment_info().collect();
 
-        let table_id = info.stream_job_fragments.stream_job_id();
-        let table_id_str = format!("{}", table_id.table_id);
-
         let actors_to_create =
             edges.collect_actors_to_create(info.stream_job_fragments.actors_to_create());
 
         let graph_info = InflightStreamingJobInfo {
-            job_id: table_id,
+            job_id,
             fragment_infos,
         };
 
         let mut barrier_control =
-            CreatingStreamingJobBarrierControl::new(table_id, backfill_epoch, false);
+            CreatingStreamingJobBarrierControl::new(job_id, backfill_epoch, false);
 
         let mut prev_epoch_fake_physical_time = 0;
         let mut pending_non_checkpoint_barriers = vec![];
 
         let initial_barrier_info = CreatingStreamingJobStatus::new_fake_barrier(
             &mut prev_epoch_fake_physical_time,
             &mut pending_non_checkpoint_barriers,
-            true,
+            PbBarrierKind::Checkpoint,
```

Comment:
Should this be `PbBarrierKind::Initial`?

Reply:
No. `Initial` is used for the very first barrier after recovery mark the beginning of stream. It indicates that we should not sync and commit on `barrier.epoch.prev`. Here the barrier kind should be the one we used for the barrier to `CreateStreamingJob`, which is `Checkpoint`.